.github/workflows/tests.yml

```diff
@@ -64,85 +62,102 @@ jobs:
     needs: changes
     if: needs.changes.outputs.code == 'true'
 
-    runs-on: windows-2019
+    runs-on: windows-latest
     strategy:
       fail-fast: false
       matrix:
         # test lower version (w/ defaults) and upper version (w/ defaults and conda-forge)
-        default-channel: ['defaults', 'conda-forge']
         python-version: ['3.8', '3.11']
-        test-type: ['unit', 'integration']
-        test-group: ['1', '2', '3']
+        default-channel: [defaults, conda-forge]
+        test-type: [unit, integration]
+        test-group: [1, 2, 3]
         exclude:
-          - default-channel: 'conda-forge'
-            python-version: '3.8'
+          - python-version: '3.8'
+            default-channel: conda-forge
+          - test-type: unit
+            test-group: 3
     env:
-      OS: Windows
-      PYTHON: ${{ matrix.python-version }}
-      TEST_SPLITS: 3
-      TEST_GROUP: ${{ matrix.test-group }}
+      ALLURE_DIR: ${{ github.workspace }}\allure-results
+      PYTEST_MARKER: ${{ matrix.test-type == 'unit' && 'not integration' || 'integration' }}
+      PYTEST_NUMPROCESSES: '0'
```

Comment:
Does this mean "automatic"?

Reply:
the docs only says this is the default value if `--pdb` is specified

looking at pytest-xdist's source it looks like a zero value means it runs inline to the spawning process, with `-n 1` it will spawn a child `popen` process to run the tests